libraries/AP_Scheduler/AP_Scheduler.tpp

```diff
@@ -0,0 +1,213 @@
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+extern const AP_HAL::HAL& hal;
+template <class Vehicle> int8_t AP_Scheduler<Vehicle>::current_task = -1;
+
+template <class Vehicle>
+const AP_Param::GroupInfo AP_Scheduler<Vehicle>::var_info[] = {
+    // @Param: DEBUG
+    // @DisplayName: Scheduler debug level
+    // @Description: Set to non-zero to enable scheduler debug messages. When set to show "Slips" the scheduler will display a message whenever a scheduled task is delayed due to too much CPU load. When set to ShowOverruns the scheduled will display a message whenever a task takes longer than the limit promised in the task table.
+    // @Values: 0:Disabled,2:ShowSlips,3:ShowOverruns
+    // @User: Advanced
+    AP_GROUPINFO("DEBUG",    0, AP_Scheduler, _debug, 0),
+
+    // @Param: LOOP_RATE
+    // @DisplayName: Scheduling main loop rate
+    // @Description: This controls the rate of the main control loop in Hz. This should only be changed by developers. This only takes effect on restart
+    // @Values: 50:50Hz,100:100Hz,200:200Hz,250:250Hz,300:300Hz,400:400Hz
+    // @RebootRequired: True
+    // @User: Advanced
+    AP_GROUPINFO("LOOP_RATE",  1, AP_Scheduler, _loop_rate_hz, 50),
+
+    AP_GROUPEND
+};
+
+// constructor
+template <class Vehicle>
+AP_Scheduler<Vehicle>::AP_Scheduler(Vehicle *parent, const uint16_t loop_rate) : _parent(parent)
+{
+    _tasks = nullptr;
+    _last_run = nullptr;
+  
+    AP_Param::setup_object_defaults(this, var_info);
+    _loop_rate_hz.set_and_save(loop_rate);
+
+    // only allow 50 to 400 Hz
+    if (_loop_rate_hz < 50) {
+        _loop_rate_hz.set(50);
+    } else if (_loop_rate_hz > 400) {
+        _loop_rate_hz.set(400);
+    }
+}
+
+// initialise the scheduler
+template <class Vehicle>
+AP_Scheduler<Vehicle>::~AP_Scheduler() {
+    if(_last_run) {
+        delete [] _last_run;
+    }
+    if(_perf_counters) {
+        delete [] _perf_counters;
+    }
+}
+
+// initialise the scheduler
+template <class Vehicle>
+void AP_Scheduler<Vehicle>::init(const AP_Task<Vehicle> *tasks, uint8_t num_tasks)
+{
+    if(tasks == nullptr || num_tasks == 0) {
+        return;
+    }
+    _tasks = tasks;
+    _num_tasks = num_tasks;
+    _last_run = new uint16_t[_num_tasks];
+    memset(_last_run, 0, sizeof(_last_run[0]) * _num_tasks);
+    _tick_counter = 0;
+}
+
+// one tick has passed
+template <class Vehicle>
+void AP_Scheduler<Vehicle>::tick(void)
+{
+    _tick_counter++;
+}
+
+/*
+  run one tick
+  this will run as many scheduler tasks as we can in the specified time
+ */
+template <class Vehicle>
+void AP_Scheduler<Vehicle>::run(uint32_t time_available)
+{
+    // break condition
+    if(_tasks == nullptr) {
+        return;
+    }
+  
+    if (_debug > 3 && _perf_counters == nullptr) {
```

Comment:
I think you got the rebase wrong. This is undoing what was done in 56aa467d60ed1af115efa57d68b6c30404d2007a. You need to change this to 1, as well as in the other places.